src/core/Song.cpp

```diff
@@ -258,16 +253,17 @@ void Song::processNextBuffer()
 		return false;
 	};
 
-	const auto timeline = getPlayPos().m_timeLine;
-	const auto loopEnabled = !m_exporting && timeline && timeline->loopPointsEnabled();
+	const auto& timeline = getTimeline();
+	const auto loopEnabled = !m_exporting && timeline.loopEnabled();
 
 	// Ensure playback begins within the loop if it is enabled
-	if (loopEnabled) { enforceLoop(timeline->loopBegin(), timeline->loopEnd()); }
+	if (loopEnabled) { enforceLoop(timeline.loopBegin(), timeline.loopEnd()); }
 
-	// Inform VST plugins if the user moved the play head
+	// Inform VST plugins and sample tracks if the user moved the play head
 	if (getPlayPos().jumped())
 	{
 		m_vstSyncController.setPlaybackJumped(true);
+		emit updateSampleTracks();
```

Comment:
This is starting to look pretty bad. Trying to think of a better way to do this.. Maybe there should be a play position jumped signal that the VST sync controller and sample tracks (and in the future, CLAP plugins) could connect to?

Reply:
I agree we should find a better way to do this (I'd particularly like to get all the VST stuff out of the core), but I think it's out of scope here. We already have the same two lines slightly earlier on in this method to handle loop points (in that case to deal with a jump triggered by the core, as opposed to here which handles jumps triggered by the user), so I don't think it's any worse than it already is.